ant-node/src/lib.rs

```diff
@@ -41,20 +45,22 @@ pub use self::{
 
 use crate::error::{Error, Result};
 
+use ant_evm::RewardsAddress;
 use ant_networking::{Network, SwarmLocalState};
 use ant_protocol::{get_port_from_multiaddr, NetworkAddress};
-use libp2p::PeerId;
+use libp2p::{Multiaddr, PeerId};
+
 use std::{
     collections::{BTreeMap, HashSet},
     path::PathBuf,
 };
-
-use ant_evm::RewardsAddress;
+use tokio::sync::watch;
 
 /// Once a node is started and running, the user obtains
 /// a `NodeRunning` object which can be used to interact with it.
 #[derive(Clone)]
 pub struct RunningNode {
+    shutdown_sender: watch::Sender<bool>,
```

Comment:
We send `TerminateNode(String)` via the `NodeEventsChannel` below, can this be re-used here?

Reply:
It could be reused for the node (`NodeEvents`) loop, but for the `SwarmDriver` loop it wouldn't make sense, as the broadcast channel uses `ant_node::event::NodeEvent`, which is inaccessible from the `ant-networking` crate in which the `SwarmDriver` loop resides.